libtiledbsoma/src/soma/managed_query.cc

```diff
@@ -1257,4 +1247,23 @@ bool ManagedQuery::_extend_and_evolve_schema<std::string>(
     }
     return false;
 }
+
+std::vector<uint8_t> ManagedQuery::_cast_bool_data(
+    ArrowSchema* schema, ArrowArray* array) {
+    if (strcmp(schema->format, "b") != 0) {
+        throw TileDBSOMAError(std::format(
+            "_cast_bit_to_uint8 expected column format to be 'b' but saw "
+            "{}",
+            schema->format));
+    }
+
+    uint8_t* data = (uint8_t*)array->buffers[1];
```

Comment:
Nit: Prefer named casts to C-style casting (for clarity + slightly less error prone). Probably `static_cast` here.